### PR TITLE
Endpoint scaling support

### PR DESCRIPTION
**Pull request by jokva**
Understand and expose the ENDSCALE keyword.

### CODE REVIEW

File: opm/parser/eclipse/EclipseState/Runspec.cpp [hunk 1]
```diff
@@ -62,23 +62,128 @@ size_t Phases::size() const noexcept {
...
+
+bool endscale_nodir( const DeckKeyword& kw ) {
+    if( kw.getRecord( 0 ).getItem( 0 ).defaultApplied( 0 ) )
+        return true;
+
+    const auto& value = kw.getRecord( 0 ).getItem( 0 ).get< std::string >( 0 );
+
+    if( value != "DIRECT" && value != "NODIR" )
+        throw std::invalid_argument(
+            "ENDSCALE argument 1 must be defaulted', 'DIR' or 'NODIR', was "
```
Comment (pgdr) [inline opm/parser/eclipse/EclipseState/Runspec.cpp]: `DIR` or `DIRECT`

File: opm/parser/eclipse/EclipseState/Runspec.cpp [hunk 2]
```diff
@@ -62,23 +62,128 @@ size_t Phases::size() const noexcept {
...
+        throw std::invalid_argument( "SCALECRS takes 'YES' or 'NO'" );
+
+    return value == 'Y';
+}
+
+bool endscale_nodir( const DeckKeyword& kw ) {
+    if( kw.getRecord( 0 ).getItem( 0 ).defaultApplied( 0 ) )
+        return true;
+
+    const auto& value = kw.getRecord( 0 ).getItem( 0 ).get< std::string >( 0 );
```
Comment (joakim-hove) [inline opm/parser/eclipse/EclipseState/Runspec.cpp]: I guess `value`should be uppercased before the comparison.

**Comment by joakim-hove** [comment]
I would like to have the `EndScale`implementation in separate files - apart from that I think this can be merged.

File: opm/parser/eclipse/EclipseState/Runspec.cpp (hunk 2 above)
Comment (jokva) [inline opm/parser/eclipse/EclipseState/Runspec.cpp]: Absolutely right.

File: opm/parser/eclipse/EclipseState/EndpointScaling.cpp [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    return bool( *this ) && !this->threepoint();
+}
+
+bool EndpointScaling::threepoint() const noexcept {
+    return this->options[ static_cast< ue >( option::threepoint ) ];
+}
+
+namespace {
+
+bool threepoint_scaling( const Deck& deck ) {
```
Comment (andlaus) [inline opm/parser/eclipse/EclipseState/EndpointScaling.cpp]: @joakim-hove: what's your plan for the naming convention? the lowercase_underscores() convention looks quite strange when mixed with camelCase() for "related" objects, IMO...

Comment (joakim-hove) [inline opm/parser/eclipse/EclipseState/EndpointScaling.cpp]: Thank you for asking - I think this slightly tricky: See #1021 for my view of the situation.

**Pull request merged**